packages/doenetml-worker-javascript/src/CoreWorker.ts

```diff
@@ -0,0 +1,320 @@
+import Core from "./Core";
+import { removeFunctionsMathExpressionClass } from "./utils/math";
+import { createComponentInfoObjects } from "./utils/componentInfoObjects";
+import {
+    addDocumentIfItsMissing,
+    expandDoenetMLsToFullSerializedComponents,
+} from "./utils/expandDoenetML";
+import { returnAllPossibleVariants } from "./utils/returnAllPossibleVariants";
+// import { dastToSerialized } from "./utils/dastToSerializedComponents";
+
+export class PublicDoenetMLCore {
```

Comment:
Docstring here as well as for each item talking about what they are for. The typescript types look unhelpful. Maybe a TODO indicating that they should be made correct? Or is this all going to be deleted soon?

Reply:
Made some docstrings as well as a TODO. This part is very much in flux.